### ISSUE TITLE
New Coroutine rule request: `runBlocking` from `main` or tests only

### ISSUE THREAD

**Issue by Munzey** [detekt/detekt#6127]
## Expected Behavior of the rule

Kotlin [documents](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/run-blocking.html) that `runBlocking` should be used 

>in main functions and in tests

All other circumstances should be discouraged and instead new coroutines ran from the relevant coroutine scope.
 
## Context

we want to avoid developers unfamiliar with coroutines from trying to use this function incorrectly.

**Comment by BraisGabin** [comment]
I agree that `runBlocking` should be discoraged. But I'm wondering if this should just be a forbidden call method. If you need it on `main` you can just `@Suppress` it.

**Comment by cortinico** [comment]
Speaking of `runBlocking`, there is also a `runTest` function that should be used instead for tests, that would make for a nice rule

**Comment by marschwar** [comment]
> Speaking of `runBlocking`, there is also a `runTest` function that should be used instead for tests, that would make for a nice rule

There already is an issue for that with #5410.
